Fix calculation of tendencies when parent atm proc is subcycled

bartgol opened the pull request:
The current tendency calculation assumed that, when an atm proc run method was called, it had the right to overwrite the content of the tend field with the start-of-step value of the field. However, this is wrong if the atm proc is part of a subcycled group. In that case, when the run method is called the second time, the tend field already contains a contribution from the previous subcycle iteration of the parent atm proc.

The simplest solution is to store the start-of-step value of the field in a separate object, to avoid overwriting the tend field.

Fixes #2583

E3SM-Autotester commented:
Status Flag 'Pre-Test Inspection' - Auto Inspected - Inspection is Not Necessary for this Pull Request.

E3SM-Autotester commented:
Status Flag 'Pull Request AutoTester' - Jenkins Testing: 1 or more Jobs FAILED

Note: Testing will normally be attempted again in approx. 2 Hrs. If a change to the PR source branch occurs, the testing will be attempted again on next available autotester run.

<details>
<summary>
  <b>Pull Request Auto Testing has FAILED</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: SCREAM_PullRequest_Autotester_Mappy</h3>
<ul>
  <li>Build Num: 4522</li>
  <li>Status: FAILED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>AT: AUTOMERGE;Atmosphere Driver;bugfix</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2584</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_SHA</td>
    <td>6227982bd7230198e6e24679c72178d032b33a5e</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_BRANCH</td>
    <td>master</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_SHA</td>
    <td>9080ef7e7245a2b8b5af812bc187e64ea066a3c6</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>SCREAM</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: SCREAM_PullRequest_Autotester_Weaver</h3>
<ul>
  <li>Build Num: 5021</li>
  <li>Status: FAILED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>AT: AUTOMERGE;Atmosphere Driver;bugfix</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2584</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_SHA</td>
    <td>6227982bd7230198e6e24679c72178d032b33a5e</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_BRANCH</td>
    <td>master</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_SHA</td>
    <td>9080ef7e7245a2b8b5af812bc187e64ea066a3c6</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>SCREAM</td>
  </tr>
</table>
</details>


<details>
<summary>
  <b>Console Output (last 100 lines) : SCREAM_PullRequest_Autotester_Mappy # 4522</b> (click to expand)
</summary>
<pre>
<code>
Starting SETUP for test PET_Ln9_P32x2.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9 with 1 procs
Finished XML for test ERP_D_Lh4.ne4_ne4.F2010-SCREAMv1.mappy_gnu9 in 1.034309 seconds (PASS)
Finished XML for test SMS_D_Ln9.ne4_ne4.F2010-SCREAMv1-noAero.mappy_gnu9 in 1.035936 seconds (PASS)
Finished XML for test ERP_Ln22.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9 in 1.041615 seconds (PASS)
Finished XML for test ERS_Ln9.ne4_ne4.F2000-SCREAMv1-AQP1.mappy_gnu9 in 1.044639 seconds (PASS)
Finished XML for test ERS_D_Ln21.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9.scream-rad_frequency_2 in 0.775481 seconds (PASS)
Starting SETUP for test ERS_D_Ln21.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9.scream-rad_frequency_2 with 1 procs
Starting SETUP for test ERP_D_Lh4.ne4_ne4.F2010-SCREAMv1.mappy_gnu9 with 1 procs
Starting SETUP for test SMS_D_Ln9.ne4_ne4.F2010-SCREAMv1-noAero.mappy_gnu9 with 1 procs
Starting SETUP for test ERS_Ln9.ne4_ne4.F2000-SCREAMv1-AQP1.mappy_gnu9 with 1 procs
Starting SETUP for test ERP_Ln22.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9 with 1 procs
Finished SETUP for test ERS_Ln9.ne4_ne4.F2000-SCREAMv1-AQP1.mappy_gnu9 in 2.525349 seconds (PASS)
Starting SHAREDLIB_BUILD for test ERS_Ln9.ne4_ne4.F2000-SCREAMv1-AQP1.mappy_gnu9 with 1 procs
Finished SETUP for test SMS_D_Ln9.ne4_ne4.F2010-SCREAMv1-noAero.mappy_gnu9 in 2.925546 seconds (PASS)
Finished SETUP for test ERP_Ln22.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9 in 2.926334 seconds (PASS)
Finished SETUP for test PET_Ln9_P32x2.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9 in 3.130303 seconds (PASS)
Finished SETUP for test ERS_D_Ln21.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9.scream-rad_frequency_2 in 2.936870 seconds (PASS)
Finished SETUP for test ERP_D_Lh4.ne4_ne4.F2010-SCREAMv1.mappy_gnu9 in 2.942899 seconds (PASS)
Starting SHAREDLIB_BUILD for test ERS_D_Ln21.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9.scream-rad_frequency_2 with 1 procs
Starting SHAREDLIB_BUILD for test ERP_D_Lh4.ne4_ne4.F2010-SCREAMv1.mappy_gnu9 with 1 procs
Starting SHAREDLIB_BUILD for test SMS_D_Ln9.ne4_ne4.F2010-SCREAMv1-noAero.mappy_gnu9 with 1 procs
Starting SHAREDLIB_BUILD for test ERP_Ln22.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9 with 1 procs
Starting SHAREDLIB_BUILD for test PET_Ln9_P32x2.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9 with 1 procs
Finished SHAREDLIB_BUILD for test ERS_D_Ln21.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9.scream-rad_frequency_2 in 41.418118 seconds (PASS)
Starting MODEL_BUILD for test ERS_D_Ln21.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9.scream-rad_frequency_2 with 16 procs
Finished SHAREDLIB_BUILD for test SMS_D_Ln9.ne4_ne4.F2010-SCREAMv1-noAero.mappy_gnu9 in 41.480329 seconds (PASS)
Starting MODEL_BUILD for test SMS_D_Ln9.ne4_ne4.F2010-SCREAMv1-noAero.mappy_gnu9 with 16 procs
Finished SHAREDLIB_BUILD for test ERS_Ln9.ne4_ne4.F2000-SCREAMv1-AQP1.mappy_gnu9 in 48.761492 seconds (PASS)
Starting MODEL_BUILD for test ERS_Ln9.ne4_ne4.F2000-SCREAMv1-AQP1.mappy_gnu9 with 16 procs
Finished SHAREDLIB_BUILD for test PET_Ln9_P32x2.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9 in 57.210369 seconds (PASS)
Starting MODEL_BUILD for test PET_Ln9_P32x2.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9 with 16 procs
Finished SHAREDLIB_BUILD for test ERP_D_Lh4.ne4_ne4.F2010-SCREAMv1.mappy_gnu9 in 128.498971 seconds (PASS)
Finished SHAREDLIB_BUILD for test ERP_Ln22.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9 in 149.143150 seconds (PASS)
Starting MODEL_BUILD for test ERP_D_Lh4.ne4_ne4.F2010-SCREAMv1.mappy_gnu9 with 16 procs
Finished MODEL_BUILD for test ERS_D_Ln21.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9.scream-rad_frequency_2 in 185.316252 seconds (PASS)
Starting RUN for test ERS_D_Ln21.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9.scream-rad_frequency_2 with 1 proc on interactive node and 64 procs on compute nodes
Finished MODEL_BUILD for test SMS_D_Ln9.ne4_ne4.F2010-SCREAMv1-noAero.mappy_gnu9 in 186.051828 seconds (PASS)
Starting RUN for test SMS_D_Ln9.ne4_ne4.F2010-SCREAMv1-noAero.mappy_gnu9 with 1 proc on interactive node and 64 procs on compute nodes
Finished RUN for test ERS_D_Ln21.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9.scream-rad_frequency_2 in 1.511424 seconds (PEND). [COMPLETED 1 of 6]
Starting MODEL_BUILD for test ERP_Ln22.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9 with 16 procs
Finished RUN for test SMS_D_Ln9.ne4_ne4.F2010-SCREAMv1-noAero.mappy_gnu9 in 1.353540 seconds (PEND). [COMPLETED 2 of 6]
Finished MODEL_BUILD for test ERS_Ln9.ne4_ne4.F2000-SCREAMv1-AQP1.mappy_gnu9 in 243.194333 seconds (PASS)
Starting RUN for test ERS_Ln9.ne4_ne4.F2000-SCREAMv1-AQP1.mappy_gnu9 with 1 proc on interactive node and 64 procs on compute nodes
Finished RUN for test ERS_Ln9.ne4_ne4.F2000-SCREAMv1-AQP1.mappy_gnu9 in 1.531314 seconds (PEND). [COMPLETED 3 of 6]
Finished MODEL_BUILD for test PET_Ln9_P32x2.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9 in 238.615669 seconds (PASS)
Starting RUN for test PET_Ln9_P32x2.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9 with 1 proc on interactive node and 64 procs on compute nodes
Finished RUN for test PET_Ln9_P32x2.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9 in 1.083576 seconds (PEND). [COMPLETED 4 of 6]
Finished MODEL_BUILD for test ERP_D_Lh4.ne4_ne4.F2010-SCREAMv1.mappy_gnu9 in 358.129718 seconds (PASS)
Starting RUN for test ERP_D_Lh4.ne4_ne4.F2010-SCREAMv1.mappy_gnu9 with 1 proc on interactive node and 64 procs on compute nodes
Finished RUN for test ERP_D_Lh4.ne4_ne4.F2010-SCREAMv1.mappy_gnu9 in 1.126832 seconds (PEND). [COMPLETED 5 of 6]
Finished MODEL_BUILD for test ERP_Ln22.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9 in 389.558490 seconds (PASS)
Starting RUN for test ERP_Ln22.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9 with 1 proc on interactive node and 64 procs on compute nodes
Finished RUN for test ERP_Ln22.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9 in 1.137330 seconds (PEND). [COMPLETED 6 of 6]
Waiting for tests to finish
PASS ERP_D_Lh4.ne4_ne4.F2010-SCREAMv1.mappy_gnu9 RUN
    Case dir: /ascldap/users/e3sm-jenkins/acme/scratch/ERP_D_Lh4.ne4_ne4.F2010-SCREAMv1.mappy_gnu9.C.20231025_124230_b4zzfb
PASS ERP_Ln22.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9 RUN
    Case dir: /ascldap/users/e3sm-jenkins/acme/scratch/ERP_Ln22.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9.C.20231025_124230_b4zzfb
PASS ERS_D_Ln21.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9.scream-rad_frequency_2 RUN
    Case dir: /ascldap/users/e3sm-jenkins/acme/scratch/ERS_D_Ln21.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9.scream-rad_frequency_2.C.20231025_124230_b4zzfb
PASS ERS_Ln9.ne4_ne4.F2000-SCREAMv1-AQP1.mappy_gnu9 RUN
    Case dir: /ascldap/users/e3sm-jenkins/acme/scratch/ERS_Ln9.ne4_ne4.F2000-SCREAMv1-AQP1.mappy_gnu9.C.20231025_124230_b4zzfb
PASS PET_Ln9_P32x2.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9 RUN
    Case dir: /ascldap/users/e3sm-jenkins/acme/scratch/PET_Ln9_P32x2.ne4pg2_ne4pg2.F2010-SCREAMv1.mappy_gnu9.C.20231025_124230_b4zzfb
PASS SMS_D_Ln9.ne4_ne4.F2010-SCREAMv1-noAero.mappy_gnu9 RUN
    Case dir: /ascldap/users/e3sm-jenkins/acme/scratch/SMS_D_Ln9.ne4_ne4.F2010-SCREAMv1-noAero.mappy_gnu9.C.20231025_124230_b4zzfb
test-scheduler took 1075.5265593528748 seconds'
+ [[ 0 != 0 ]]
+ set +x
######################################################
FAILS DETECTED:
  SCREAM STANDALONE TESTING FAILED!
Build type full_debug failed at testing time. Here's a list of failed tests:
264:p3_tend_check_np1
265:p3_tend_check_np2
266:p3_tend_check_np3
267:p3_tend_check_np4
275:shoc_tend_check_np1
276:shoc_tend_check_np2
277:shoc_tend_check_np3
278:shoc_tend_check_np4
362:homme_shoc_cld_spa_p3_rrtmgp_128levels_tend_check_np1
363:homme_shoc_cld_spa_p3_rrtmgp_128levels_tend_check_np2
364:homme_shoc_cld_spa_p3_rrtmgp_128levels_tend_check_np3
365:homme_shoc_cld_spa_p3_rrtmgp_128levels_tend_check_np4
######################################################
Build step 'Execute shell' marked build as failure
Performing Post build task...
Match found for : : True
Logical operation result is TRUE
Running script  : #!/bin/bash -le

cd $WORKSPACE/${BUILD_ID}/

./scream/components/eamxx/scripts/jenkins/jenkins_cleanup.sh
[SCREAM_PullRequest_Autotester_Mappy] $ /bin/bash -le /tmp/jenkins14455513832677779431.sh
POST BUILD TASK : SUCCESS
END OF POST BUILD TASK : 0
Sending e-mails to: lbertag@sandia.gov
Finished: FAILURE

</code>
</pre>
</details>

<details>
<summary>
  <b>Console Output (last 100 lines) : SCREAM_PullRequest_Autotester_Weaver # 5021</b> (click to expand)
</summary>
<pre>
<code>
75:p3_tend_check_np1
77:shoc_tend_check_np1
107:homme_shoc_cld_spa_p3_rrtmgp_128levels_tend_check_np1

Build type full_sp_debug failed at testing time. Here'\''s a list of failed tests:
69:p3_tend_check_np1
71:shoc_tend_check_np1

Build type release failed at testing time. Here'\''s a list of failed tests:
74:p3_tend_check_np1
76:shoc_tend_check_np1
106:homme_shoc_cld_spa_p3_rrtmgp_128levels_tend_check_np1

Error(s) occurred during test phase
OVERALL STATUS: FAIL
Starting analysis on weaver with cmd: cd /home/e3sm-jenkins/weaver/workspace/SCREAM_PullRequest_Autotester_Weaver/5021/scream/components/eamxx && source /etc/profile.d/modules.sh && module purge && module load cmake/3.25.1 git/2.39.1 python/3.10.8 py-netcdf4/1.5.8 gcc/11.3.0 cuda/11.8.0 openmpi netcdf-c netcdf-fortran parallel-netcdf netlib-lapack && true && bsub -I -q rhel8 -n 4 -gpu num=4 ./scripts/test-all-scream --baseline-dir AUTO $compiler -c EKAT_DISABLE_TPL_WARNINGS=ON -p -i -m weaver
RUN: cd /home/e3sm-jenkins/weaver/workspace/SCREAM_PullRequest_Autotester_Weaver/5021/scream/components/eamxx && source /etc/profile.d/modules.sh && module purge && module load cmake/3.25.1 git/2.39.1 python/3.10.8 py-netcdf4/1.5.8 gcc/11.3.0 cuda/11.8.0 openmpi netcdf-c netcdf-fortran parallel-netcdf netlib-lapack && true && bsub -I -q rhel8 -n 4 -gpu num=4 ./scripts/test-all-scream --baseline-dir AUTO $compiler -c EKAT_DISABLE_TPL_WARNINGS=ON -p -i -m weaver
FROM: /home/e3sm-jenkins/weaver/workspace/SCREAM_PullRequest_Autotester_Weaver/5021/scream/components/eamxx
weaver failed'
+ errors='Build type full_debug failed at testing time. Here'\''s a list of failed tests:
75:p3_tend_check_np1
77:shoc_tend_check_np1
107:homme_shoc_cld_spa_p3_rrtmgp_128levels_tend_check_np1

Build type full_sp_debug failed at testing time. Here'\''s a list of failed tests:
69:p3_tend_check_np1
71:shoc_tend_check_np1

Build type release failed at testing time. Here'\''s a list of failed tests:
74:p3_tend_check_np1
76:shoc_tend_check_np1
106:homme_shoc_cld_spa_p3_rrtmgp_128levels_tend_check_np1

Error(s) occurred during test phase
OVERALL STATUS: FAIL
Starting analysis on weaver with cmd: cd /home/e3sm-jenkins/weaver/workspace/SCREAM_PullRequest_Autotester_Weaver/5021/scream/components/eamxx && source /etc/profile.d/modules.sh && module purge && module load cmake/3.25.1 git/2.39.1 python/3.10.8 py-netcdf4/1.5.8 gcc/11.3.0 cuda/11.8.0 openmpi netcdf-c netcdf-fortran parallel-netcdf netlib-lapack && true && bsub -I -q rhel8 -n 4 -gpu num=4 ./scripts/test-all-scream --baseline-dir AUTO $compiler -c EKAT_DISABLE_TPL_WARNINGS=ON -p -i -m weaver
RUN: cd /home/e3sm-jenkins/weaver/workspace/SCREAM_PullRequest_Autotester_Weaver/5021/scream/components/eamxx && source /etc/profile.d/modules.sh && module purge && module load cmake/3.25.1 git/2.39.1 python/3.10.8 py-netcdf4/1.5.8 gcc/11.3.0 cuda/11.8.0 openmpi netcdf-c netcdf-fortran parallel-netcdf netlib-lapack && true && bsub -I -q rhel8 -n 4 -gpu num=4 ./scripts/test-all-scream --baseline-dir AUTO $compiler -c EKAT_DISABLE_TPL_WARNINGS=ON -p -i -m weaver
FROM: /home/e3sm-jenkins/weaver/workspace/SCREAM_PullRequest_Autotester_Weaver/5021/scream/components/eamxx
weaver failed'
+ SA_FAILURES_DETAILS+='Build type full_debug failed at testing time. Here'\''s a list of failed tests:
75:p3_tend_check_np1
77:shoc_tend_check_np1
107:homme_shoc_cld_spa_p3_rrtmgp_128levels_tend_check_np1

Build type full_sp_debug failed at testing time. Here'\''s a list of failed tests:
69:p3_tend_check_np1
71:shoc_tend_check_np1

Build type release failed at testing time. Here'\''s a list of failed tests:
74:p3_tend_check_np1
76:shoc_tend_check_np1
106:homme_shoc_cld_spa_p3_rrtmgp_128levels_tend_check_np1

Error(s) occurred during test phase
OVERALL STATUS: FAIL
Starting analysis on weaver with cmd: cd /home/e3sm-jenkins/weaver/workspace/SCREAM_PullRequest_Autotester_Weaver/5021/scream/components/eamxx && source /etc/profile.d/modules.sh && module purge && module load cmake/3.25.1 git/2.39.1 python/3.10.8 py-netcdf4/1.5.8 gcc/11.3.0 cuda/11.8.0 openmpi netcdf-c netcdf-fortran parallel-netcdf netlib-lapack && true && bsub -I -q rhel8 -n 4 -gpu num=4 ./scripts/test-all-scream --baseline-dir AUTO $compiler -c EKAT_DISABLE_TPL_WARNINGS=ON -p -i -m weaver
RUN: cd /home/e3sm-jenkins/weaver/workspace/SCREAM_PullRequest_Autotester_Weaver/5021/scream/components/eamxx && source /etc/profile.d/modules.sh && module purge && module load cmake/3.25.1 git/2.39.1 python/3.10.8 py-netcdf4/1.5.8 gcc/11.3.0 cuda/11.8.0 openmpi netcdf-c netcdf-fortran parallel-netcdf netlib-lapack && true && bsub -I -q rhel8 -n 4 -gpu num=4 ./scripts/test-all-scream --baseline-dir AUTO $compiler -c EKAT_DISABLE_TPL_WARNINGS=ON -p -i -m weaver
FROM: /home/e3sm-jenkins/weaver/workspace/SCREAM_PullRequest_Autotester_Weaver/5021/scream/components/eamxx
weaver failed'
+ [[ 1 == 0 ]]
+ [[ weaver == \m\a\p\p\y ]]
+ set +x
######################################################
FAILS DETECTED:
  SCREAM STANDALONE TESTING FAILED!
Build type full_debug failed at testing time. Here's a list of failed tests:
75:p3_tend_check_np1
77:shoc_tend_check_np1
107:homme_shoc_cld_spa_p3_rrtmgp_128levels_tend_check_np1

Build type full_sp_debug failed at testing time. Here's a list of failed tests:
69:p3_tend_check_np1
71:shoc_tend_check_np1

Build type release failed at testing time. Here's a list of failed tests:
74:p3_tend_check_np1
76:shoc_tend_check_np1
106:homme_shoc_cld_spa_p3_rrtmgp_128levels_tend_check_np1

Error(s) occurred during test phase
OVERALL STATUS: FAIL
Starting analysis on weaver with cmd: cd /home/e3sm-jenkins/weaver/workspace/SCREAM_PullRequest_Autotester_Weaver/5021/scream/components/eamxx && source /etc/profile.d/modules.sh && module purge && module load cmake/3.25.1 git/2.39.1 python/3.10.8 py-netcdf4/1.5.8 gcc/11.3.0 cuda/11.8.0 openmpi netcdf-c netcdf-fortran parallel-netcdf netlib-lapack && true && bsub -I -q rhel8 -n 4 -gpu num=4 ./scripts/test-all-scream --baseline-dir AUTO $compiler -c EKAT_DISABLE_TPL_WARNINGS=ON -p -i -m weaver
RUN: cd /home/e3sm-jenkins/weaver/workspace/SCREAM_PullRequest_Autotester_Weaver/5021/scream/components/eamxx && source /etc/profile.d/modules.sh && module purge && module load cmake/3.25.1 git/2.39.1 python/3.10.8 py-netcdf4/1.5.8 gcc/11.3.0 cuda/11.8.0 openmpi netcdf-c netcdf-fortran parallel-netcdf netlib-lapack && true && bsub -I -q rhel8 -n 4 -gpu num=4 ./scripts/test-all-scream --baseline-dir AUTO $compiler -c EKAT_DISABLE_TPL_WARNINGS=ON -p -i -m weaver
FROM: /home/e3sm-jenkins/weaver/workspace/SCREAM_PullRequest_Autotester_Weaver/5021/scream/components/eamxx
weaver failed
######################################################
Build step 'Execute shell' marked build as failure
Performing Post build task...
Match found for : : True
Logical operation result is TRUE
Running script  : #!/bin/bash -le

cd $WORKSPACE/${BUILD_ID}/

./scream/components/eamxx/scripts/jenkins/jenkins_cleanup.sh
[SCREAM_PullRequest_Autotester_Weaver] $ /bin/bash -le /tmp/jenkins15681565669185411812.sh
POST BUILD TASK : SUCCESS
END OF POST BUILD TASK : 0
Sending e-mails to: lbertag@sandia.gov
Finished: FAILURE

</code>
</pre>
</details>

E3SM-Autotester commented:
Status Flag 'Pre-Test Inspection' - Auto Inspected - Inspection is Not Necessary for this Pull Request.

E3SM-Autotester commented:
Status Flag 'Pull Request AutoTester' - Testing Jenkins Projects:

<details>
<summary>
  <b>Pull Request Auto Testing STARTING</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: SCREAM_PullRequest_Autotester_Mappy</h3>
<ul>
  <li>Build Num: 4524</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>AT: AUTOMERGE;Atmosphere Driver;bugfix</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2584</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_SHA</td>
    <td>0b796eea78fe21a0e9c4e67f9f836f713978df8a</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_BRANCH</td>
    <td>master</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_SHA</td>
    <td>9080ef7e7245a2b8b5af812bc187e64ea066a3c6</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>SCREAM</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: SCREAM_PullRequest_Autotester_Weaver</h3>
<ul>
  <li>Build Num: 5023</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>AT: AUTOMERGE;Atmosphere Driver;bugfix</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2584</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_SHA</td>
    <td>0b796eea78fe21a0e9c4e67f9f836f713978df8a</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_BRANCH</td>
    <td>master</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_SHA</td>
    <td>9080ef7e7245a2b8b5af812bc187e64ea066a3c6</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>SCREAM</td>
  </tr>
</table>

<h2>Using Repos:</h2>

<dl>
  <dt>Repo: SCREAM (E3SM-Project/scream)</dt>
  <ul>
    <li>Branch: bartgol/fix-2583</li>
    <li>SHA: 0b796eea78fe21a0e9c4e67f9f836f713978df8a</li>
    <li>Mode: TEST_REPO</li>
  </ul>
</dl>

Pull Request Author: bartgol
</details>

E3SM-Autotester commented:
Status Flag 'Pull Request AutoTester' - Jenkins Testing: all Jobs PASSED

<details>
<summary>
  <b>Pull Request Auto Testing has PASSED</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: SCREAM_PullRequest_Autotester_Mappy</h3>
<ul>
  <li>Build Num: 4524</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>AT: AUTOMERGE;Atmosphere Driver;bugfix</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2584</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_SHA</td>
    <td>0b796eea78fe21a0e9c4e67f9f836f713978df8a</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_BRANCH</td>
    <td>master</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_SHA</td>
    <td>9080ef7e7245a2b8b5af812bc187e64ea066a3c6</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>SCREAM</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: SCREAM_PullRequest_Autotester_Weaver</h3>
<ul>
  <li>Build Num: 5023</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>AT: AUTOMERGE;Atmosphere Driver;bugfix</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2584</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_SHA</td>
    <td>0b796eea78fe21a0e9c4e67f9f836f713978df8a</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_BRANCH</td>
    <td>master</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_SHA</td>
    <td>9080ef7e7245a2b8b5af812bc187e64ea066a3c6</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>SCREAM</td>
  </tr>
</table>
</details>

E3SM-Autotester commented:
Status Flag 'Pre-Merge Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           NO REVIEWS HAVE BEEN PERFORMED ON THIS PULL REQUEST!

E3SM-Autotester commented:
All Jobs Finished; status = PASSED, However Inspection must be performed before merge can occur...

AaronDonahue reviewed (changes requested):
Looks good.  My only question would be, is there a way to create a test (or add to an existing one) to check for proper accumulation of tendencies when subcycling?

E3SM-Autotester commented:
Status Flag 'Pre-Merge Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           THE LAST COMMIT TO THIS PULL REQUEST HAS BEEN REVIEWED, BUT NOT ACCEPTED OR REQUIRES CHANGES!

E3SM-Autotester commented:
All Jobs Finished; status = PASSED, However Inspection must be performed before merge can occur...

bartgol commented:
> Looks good. My only question would be, is there a way to create a test (or add to an existing one) to check for proper accumulation of tendencies when subcycling?

Thanks for making me look into this. Turned out that doing `t+=(fnew-fold)/dt` from within the proc yielded the wrong tendency (dt was dt/N in case of subcycling). I fixed it, and added a tend check test in the shoc_p3 subcycling test.

E3SM-Autotester commented:
Status Flag 'Pre-Test Inspection' - Auto Inspected - Inspection is Not Necessary for this Pull Request.

E3SM-Autotester commented:
Status Flag 'Pull Request AutoTester' - Testing Jenkins Projects:

<details>
<summary>
  <b>Pull Request Auto Testing STARTING</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: SCREAM_PullRequest_Autotester_Mappy</h3>
<ul>
  <li>Build Num: 4526</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>AT: AUTOMERGE;Atmosphere Driver;bugfix</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2584</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_SHA</td>
    <td>2e9f05d1da157a5a170bb08e40312d6026a3ad28</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_BRANCH</td>
    <td>master</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_SHA</td>
    <td>9080ef7e7245a2b8b5af812bc187e64ea066a3c6</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>SCREAM</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: SCREAM_PullRequest_Autotester_Weaver</h3>
<ul>
  <li>Build Num: 5025</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>AT: AUTOMERGE;Atmosphere Driver;bugfix</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2584</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_SHA</td>
    <td>2e9f05d1da157a5a170bb08e40312d6026a3ad28</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_BRANCH</td>
    <td>master</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_SHA</td>
    <td>9080ef7e7245a2b8b5af812bc187e64ea066a3c6</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>SCREAM</td>
  </tr>
</table>

<h2>Using Repos:</h2>

<dl>
  <dt>Repo: SCREAM (E3SM-Project/scream)</dt>
  <ul>
    <li>Branch: bartgol/fix-2583</li>
    <li>SHA: 2e9f05d1da157a5a170bb08e40312d6026a3ad28</li>
    <li>Mode: TEST_REPO</li>
  </ul>
</dl>

Pull Request Author: bartgol
</details>

E3SM-Autotester commented:
Status Flag 'Pull Request AutoTester' - Jenkins Testing: all Jobs PASSED

<details>
<summary>
  <b>Pull Request Auto Testing has PASSED</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: SCREAM_PullRequest_Autotester_Mappy</h3>
<ul>
  <li>Build Num: 4526</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>AT: AUTOMERGE;Atmosphere Driver;bugfix</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2584</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_SHA</td>
    <td>2e9f05d1da157a5a170bb08e40312d6026a3ad28</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_BRANCH</td>
    <td>master</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_SHA</td>
    <td>9080ef7e7245a2b8b5af812bc187e64ea066a3c6</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>SCREAM</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: SCREAM_PullRequest_Autotester_Weaver</h3>
<ul>
  <li>Build Num: 5025</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>AT: AUTOMERGE;Atmosphere Driver;bugfix</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>2584</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_SHA</td>
    <td>2e9f05d1da157a5a170bb08e40312d6026a3ad28</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_BRANCH</td>
    <td>master</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_SHA</td>
    <td>9080ef7e7245a2b8b5af812bc187e64ea066a3c6</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>SCREAM</td>
  </tr>
</table>
</details>

E3SM-Autotester commented:
Status Flag 'Pre-Merge Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           THE LAST COMMIT TO THIS PULL REQUEST HAS NOT BEEN REVIEWED YET!

E3SM-Autotester commented:
All Jobs Finished; status = PASSED, However Inspection must be performed before merge can occur...

E3SM-Autotester commented:
Status Flag 'Pre-Merge Inspection' - SUCCESS: The last commit to this Pull Request has been INSPECTED AND APPROVED by [ AaronDonahue ]!

E3SM-Autotester commented:
Status Flag 'Pull Request AutoTester' - Pull Request will be Automerged

E3SM-Autotester commented:
Merge on Pull Request# 2584: IS A SUCCESS - Pull Request successfully merged
